### PR TITLE
cleanup: remove obsolete option "filestore_xattr_use_omap"

### PR DESCRIPTION
**Pull request by vumrao**
This patch does cleanup for option "filestore_xattr_use_omap",
as this option was removed in #7408.

Fixes: #14356
Signed-off-by: Vikhyat Umrao <vumrao@redhat.com>

### CODE REVIEW

File: src/test/objectstore/store_test.cc [hunk 1]
```diff
@@ -3296,7 +3296,6 @@ TEST(EXT4StoreTest, _detect_fs) {
   // without user_xattr, ext4 fails
   //
   {
-    g_ceph_context->_conf->set_val("filestore_xattr_use_omap", "true");
```
Comment (tchaikov) [inline src/test/objectstore/store_test.cc]: @dachary what do you think about the change regarding to this test?

shall we just remove it?

File: src/test/xattr_bench.cc [hunk 2]
```diff
@@ -152,15 +152,6 @@ int main(int argc, char **argv) {
   argv_to_vec(argc, (const char **)argv, args);
 
   global_init(0, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_UTILITY, 0);
-  common_init_finish(g_ceph_context);
```
Comment (tchaikov) [inline src/test/xattr_bench.cc]: @vumrao why are you removing this line?

**Comment by tchaikov** [comment]
@vumrao you might want to change `Cleanup #14356` to `Fixes: #14356` in your commit message.

File: src/test/xattr_bench.cc (hunk 2 above)
Comment (vumrao) [inline src/test/xattr_bench.cc]: @tchaikov  I checked we are using "g_ceph_context" only for accessing "filestore_xattr_use_omap" and now we are removing it so thought no need to initialize it.

**Comment by vumrao** [comment]
@tchaikov Thank you Kefu for your review , sure I will change Cleanup #14356 to Fixes: #14356

File: src/test/xattr_bench.cc (hunk 2 above)
Comment (vumrao) [inline src/test/xattr_bench.cc]: @tchaikov am I missing something ?

File: src/test/objectstore/store_test.cc (hunk 1 above)
Comment (dachary) [inline src/test/objectstore/store_test.cc]: yes.

File: src/test/xattr_bench.cc (hunk 2 above)
Comment (tchaikov) [inline src/test/xattr_bench.cc]: as a convention, we call `common_init_finish()` after `global_init()` to finish the initialization. as the accompanied comment puts

>  * This includes things like starting threads, initializing libraries that
>  * can't handle forking, and so forth.

and if we take it as a black box, we should not assume what it is doing under the hood. the `g_ceph_context`could be used somewhere else by ObjectStore in some unknown way. so we'd better leave as it is even we know ObjectStore does not use anything initialized by `common_init_finish()`.

Comment (vumrao) [inline src/test/xattr_bench.cc]: Thank You @tchaikov for your nice explanation, got your point. Will change it back.

File: src/test/objectstore/store_test.cc (hunk 1 above)
Comment (vumrao) [inline src/test/objectstore/store_test.cc]: @dachary Thanks !

**Comment by vumrao** [comment]
@tchaikov thanks, I have sent next patch set after modifying suggested changes.

**Comment by dachary** [comment]
LGTM !

**Comment by tchaikov** [comment]
lgtm.

**Comment by vumrao** [comment]
@liewegas thank you!

**Pull request merged**